### ISSUE TITLE
[8.0] could not obtain lock on row in relation "ir_cron"

### ISSUE THREAD

**Issue by mathi123** [odoo/odoo#5459]
In my postgresql log, I see following error (keeps on being logged over and over):

2015-01-28 17:40:26 UTC ERROR:  could not obtain lock on row in relation "ir_cron"
2015-01-28 17:40:26 UTC STATEMENT:  SELECT *
                                       FROM ir_cron
                                       WHERE numbercall != 0
                                          AND active
                                          AND nextcall <= (now() at time zone 'UTC')
                                          AND id=2
                                       FOR UPDATE NOWAIT
2015-01-28 17:41:32 UTC ERROR:  could not obtain lock on row in relation "ir_cron"
2015-01-28 17:41:32 UTC STATEMENT:  SELECT *
                                       FROM ir_cron
                                       WHERE numbercall != 0
                                          AND active
                                          AND nextcall <= (now() at time zone 'UTC')
                                          AND id=7
                                       FOR UPDATE NOWAIT
2015-01-28 18:36:39 UTC ERROR:  could not obtain lock on row in relation "ir_cron"
....

What is causing this error?

I have following configuration:
Odoo v8.0
Ubuntu 14.04 LTS on vps
No outgoing mailserver configured

Potentially relevant parts of my config file:
limit_memory_hard = 2684354560
limit_memory_soft = 2147483648
limit_request = 8192
limit_time_cpu = 1200
limit_time_real = 120
workers = 5

**Comment by c-goosen** [comment]
This might help. 
https://www.odoo.com/es_ES/forum/help-1/question/could-not-obtain-lock-on-row-in-relation-ir-cron-74519

**Comment by farazda1** [comment]
set following value in odoo config file to 1
CURRENT
max_cron_threads = 2
SHOULD BE
max_cron_threads = 1

**Comment by bistaray** [comment]
Any other suggestions?  I have:

> max_cron_threads = 1

in the config file.

**Comment by Danultimate** [comment]
Im having the same I set the max_cron_threads = 1 But Im still having the errors. any ideas?

**Comment by Danultimate** [comment]
@bistaray Try to restart your postgresql service, that works for me.

**Comment by Yenthe666** [comment]
Hi @mathi123,

Thank you for your bug report and our apoligies that we haven't had the time to look at this bug report. We do our best to handle all of them, but we sometimes miss some.
At this point Odoo 8 is no longer supported as Odoo only supports three versions (9.0, 10.0, 11.0 at the moment).
If you can still reproduce this issue on Odoo 9, Odoo 10 or Odoo 11 we encourage you to create a new bug report.

Regards,
Yenthe

**Comment by Sarah-JJ** [comment]
In my case, it turned out there was a cron that a previous developer added to the system that was always running and never getting done. I inspected the Odoo log file for ir_cron logs. A pair of starting/done logs look like so:
2024-05-30 10:09:29,891 481994 INFO HalaFTTH_db odoo.addons.base.models.ir_cron: Starting job `Jobs Garbage Collector`.
2024-05-30 10:09:29,895 481994 INFO HalaFTTH_db odoo.addons.base.models.ir_cron: Job `Jobs Garbage Collector` done.
However, my long running cron only had a 'Starting' but never a 'done' log. When I turned off that cron, the module upgraded normally.
